src/main/java/gameframework/motion/MoveStrategyKeyboard8Dir.java

```diff
@@ -0,0 +1,92 @@
+package gameframework.motion;
+
+import java.awt.Point;
+import java.awt.event.KeyEvent;
+
+/**
+ * {@link MoveStrategy} which listens to the keyboard and answers new
+ * {@link SpeedVector speed vectors} based on what the user typed. It is
+ * possible to move to 8 directions. It is not possible to continue moving when
+ * the keys are released
+ * 
+ * @author Arnaud Cojez
+ */
+public class MoveStrategyKeyboard8Dir extends MoveStrategyKeyboard {
+
+	// Methods
+
+	/**
+	 * Constructor for the MoveStrategyKeyboard8Dir class
+	 */
+	public MoveStrategyKeyboard8Dir() {
+		this(new SpeedVector(new Point(0, 0)));
+	}
+
+	/**
+	 * Constructor for the MoveStrategyKeyboard8Dir class
+	 * 
+	 * @param speedVector
+	 *            the speedVector we want to set to the Strategy
+	 */
+	public MoveStrategyKeyboard8Dir(SpeedVector speedVector) {
+		super(false, speedVector);
```

Comment:
same problem: This constructor call is problematic because it forces the `alwaysMove` boolean to be `true`. But, as a subclass, you should let the superclass decide about the default value of its parameters. Please call the closest super constructor instead.